tests/functional/lib/helpers.js

```diff
@@ -160,8 +236,24 @@ define([
    * @returns {promise} rejects if element does not exist
    */
   const testElementExists = thenify(function (selector) {
+    var findError;
     return this.parent
       .findByCssSelector(selector)
+      .then(null, function (err) {
+        // swallow the error until after taking a screenshot.
+        // The error has to be swallowed before a screenshot
+        // can be taken or else takeScreenshot itself returns
+        // the error.
```

Comment:
Yeah, I was surprised I had to do this too. If I did a `return this.parent.takeScreenshot()` from within the error handler, `this.parent.takeScreenshot()` always rejects with the same error that the error handler was called with. To avoid that, I swallow the error, take the screenshot, then re-throw the error.